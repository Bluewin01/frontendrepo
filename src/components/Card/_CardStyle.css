.card {
  display: flex;
  margin: 20px 10px 10px;
  border-radius: 10px;
  border: solid rgb(219, 219, 219) 0.5px;
  width: 160px;
  height: 160px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
}

.card:hover {
  box-shadow: 0 8px 12px 0 rgba(0, 0, 0, 0.2);
  cursor: pointer;
}

.cardContent {
  padding: 5px;
  margin: auto;
  display: flex;
  text-align: center;
}

.cardButton {
  display: flex;
  width: auto;
  border-radius: 10px;
  border-bottom: solid rgb(219, 219, 219) 0.5px;
  justify-content: center;
  align-items: center;
}

.cardButton:hover {
  background-color: #e6edeb;
  cursor: pointer;
}

.cardButton label {
  display: flex;
  align-items: center;
  margin: 0px auto 0px;
}

.hiddenButton {
  display: none;
}

.iconDesign {
  font-size: 150%;
  color: rgb(180, 180, 180);
}

.icon {
  margin-right: 5px;
}
