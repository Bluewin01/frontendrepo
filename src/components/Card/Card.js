import React, { useState } from "react";
import "./_CardStyle.css";
import { DeleteOutlined, EditOutlined, PlusOutlined } from "@ant-design/icons";

export const Card = (props) => {
  const { cardType } = props;
  const [isHovered, setisHovered] = useState(false);

  return (() => {
    switch (cardType) {
      case "createTemplate":
        const { createTemplate } = props;
        return (
          <div
            data-testid="createTemplate"
            className="card"
            onClick={createTemplate}
          >
            <div className="cardContent">
              <PlusOutlined className="iconDesign" />
            </div>
          </div>
        );
      case "templateList":
        const { template, buttons } = props;

        const ButtonSize = () => {
          const buttonSize = 160 / buttons.length;
          return buttonSize;
        };

        return (
          <div
            key={template.id}
            className="card"
            onMouseEnter={() => {
              setisHovered(true);
            }}
            onMouseLeave={() => {
              setisHovered(false);
            }}
          >
            <div
              className="cardContent"
              style={{
                display: isHovered ? "none" : "block",
              }}
            >
              {template.name}
            </div>
            <div
              className="hiddenButton"
              style={{
                display: isHovered ? "block" : "none",
              }}
            >
              {buttons.map((button) => {
                return (
                  <div
                    key={button.name}
                    data-testid="editTemplate"
                    onClick={() => {
                      button.onClick(template.id);
                    }}
                    className="cardButton"
                    style={{ height: `${ButtonSize()}px` }}
                  >
                    {button.name === "Edit" ? (
                      <EditOutlined className="icon" />
                    ) : null}
                    {button.name === "Delete" ? (
                      <DeleteOutlined className="icon" />
                    ) : null}
                    {button.name}
                  </div>
                );
              })}
            </div>
          </div>
        );
      default:
        return null;
    }
  })();
};
