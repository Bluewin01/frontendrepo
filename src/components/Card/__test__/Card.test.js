import React from "react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { Card } from "../Card";

describe("Component: Card", () => {
  it("card renders correctly", () => {
    const tree = renderer.create(<Card></Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
