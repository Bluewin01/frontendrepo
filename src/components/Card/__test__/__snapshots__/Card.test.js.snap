// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Card card renders correctly 1`] = `null`;
